src/parser/parse_redirection.c

```diff
@@ -26,19 +27,31 @@ int	get_file_name_and_length(char *buffer, char *input)
 	}
 	else
 		len_to_cpy = file_name_len;
-	ft_strlcpy(&buffer[0], cursor, len_to_cpy + 1);
+	ft_strlcpy(&buffer[0], cursor, len_to_cpy);
 	return (file_name_len);
 }
 
-/*
-	Might still split this into diff functions
-*/
+int	get_redirect_id(const char *cursor)
+{
+	if (*cursor == LEFT_ANGLE && *(cursor + 1) != LEFT_ANGLE)
+		return (LEFT_ANGLE);
+	else if (*cursor == RIGHT_ANGLE && *(cursor + 1) != RIGHT_ANGLE)
+		return (FT_TRUNCATE);
+	else if (ft_strncmp(cursor, "<<", 2) == 0)
+		return (HERE_DOC);
+	else if (ft_strncmp(cursor, ">>", 2) == 0)
+		return (FT_APPEND);
+	return (-1);
+}
```

Comment:
If you change the order i think you don't have to check the next char, could you try?
```C

if (ft_strncmp(cursor, "<<", 2) == 0)
	return (HERE_DOC);
else if (ft_strncmp(cursor, ">>", 2) == 0)
	return (FT_APPEND);
else if (*cursor == LEFT_ANGLE)
	return (LEFT_ANGLE);
else if (*cursor == RIGHT_ANGLE)
	return (FT_TRUNCATE);
return (INVALID);
```
Also think we should use a macro instead of -1, I think we have a INVALID defined for that.

Reply:
Nice spot!
Also added one more check for an edge case `<>`